### PR TITLE
[Internal] Pipelines: Adds yml Files for New Multi-region CI Pipelines

### PR DESCRIPTION
**Pull request by NaluTripician**
# Pull Request Template

## Description

This PR adds the yml files needed to run the new multi-region ci pipelines which will help improve test coverage for the SDK. 

## Type of change

Please delete options that are not relevant.

- [] New feature (non-breaking change which adds functionality)

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: azure-pipelines-multiregion-multimaster.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    - 'docs/**/*'
+    - 'Microsoft.Azure.Cosmos/contracts/**/*'
+
+variables:
+  DebugArguments: ' --filter "TestCategory==MultiRegionMultiMaster" --verbosity normal '
+  ReleaseArguments: ' --filter "TestCategory!=MultiRegionMultiMaster" --verbosity normal '
+  VmImage: windows-latest # https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops
+
+
+jobs:
```
Comment (kirankumarkolli) [inline azure-pipelines-multiregion-multimaster.yml:23]: Conceptually assuming only the Arguments are changing then can they be parameterized? (then for multi master we can override them?

File: azure-pipelines-multiregion.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  paths:
+    include:
+    - '*'
+    exclude:
+    - '*.md'
+    - 'docs/**/*'
+    - 'Microsoft.Azure.Cosmos/contracts/**/*'
+
+variables:
+  DebugArguments: ' --filter "TestCategory==$(TestCategory)" --verbosity normal '
```
Comment (kirankumarkolli) [inline azure-pipelines-multiregion.yml:18]: Are all tests targeted for MM or a sub-set?

File: azure-pipelines-multiregion.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# A pipeline with no CI trigger
+trigger: none
+
+pr:
+  branches:
+    include:
+    - master
+    - releases/*
```
Comment (ealsur) [inline azure-pipelines-multiregion.yml:8]: Will it run on releases too?

File: azure-pipelines-multiregion.yml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    - 'Microsoft.Azure.Cosmos/contracts/**/*'
+
+variables:
+  DebugArguments: ' --filter "TestCategory==$(TestCategory)" --verbosity normal '
+  ReleaseArguments: ' --filter "TestCategory!=$(TestCategory)" --verbosity normal '
+  VmImage: windows-latest # https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops
+
+
+jobs:
+- template:  templates/static-tools.yml
```
Comment (ealsur) [inline azure-pipelines-multiregion.yml:24]: Why do we need static tool analysis?

File: azure-pipelines-multiregion.yml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    VmImage: $(VmImage)
+
+- template:  templates/nuget-pack.yml
+  parameters:
+    BuildConfiguration: Release
+    VmImage: $(VmImage)
+    ReleasePackage: false
+    OutputPath: '$(Build.ArtifactStagingDirectory)/bin/AnyCPU/Release/Microsoft.Azure.Cosmos'
+
+- template:  templates/build-ctl.yml
```
Comment (ealsur) [inline azure-pipelines-multiregion.yml:36]: Same question for below sections:

* build-ctl - This should already be verified on the normal CI
* build-samples - Same
* build-benchmark - Same
* build-internal - Same
* build-preview - Same

File: azure-pipelines-multiregion.yml [hunk 6]
```diff
@@ -0,0 +1,70 @@
+# A pipeline with no CI trigger
```
Comment (ealsur) [inline azure-pipelines-multiregion.yml:1]: When will this CI be triggered? Per PR? Or nightly? or every time a PR gets merged on master?

File: azure-pipelines-multiregion.yml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    include:
+    - '*'
+    exclude:
+    - '*.md'
+    - 'docs/**/*'
+    - 'Microsoft.Azure.Cosmos/contracts/**/*'
+
+variables:
+  DebugArguments: ' --filter "TestCategory==$(TestCategory)" --verbosity normal '
+  ReleaseArguments: ' --filter "TestCategory!=$(TestCategory)" --verbosity normal '
```
[the remark this answers is not in the document]
Comment (NaluTripician) [reply azure-pipelines-multiregion.yml]: Clarified offline

File: azure-pipelines-multiregion.yml (hunk 3 above)
Comment (NaluTripician) [reply azure-pipelines-multiregion.yml]: Will run on every PR/release

File: templates/build-test.yml [hunk 8]
```diff
@@ -5,10 +5,13 @@ parameters:
   Arguments: ''
   VmImage: '' # https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops
   OS: 'Windows'
-  EmulatorPipeline1Arguments: ' --filter "TestCategory!=Quarantine & TestCategory!=Functional & TestCategory!=ClientTelemetryRelease & TestCategory!=LongRunning & (TestCategory=ClientTelemetryEmulator|TestCategory=Query|TestCategory=ReadFeed|TestCategory=Batch|TestCategory=ChangeFeed)" --verbosity normal '
-  EmulatorPipeline2Arguments: ' --filter "TestCategory!=Quarantine & TestCategory!=Functional & TestCategory!=ClientTelemetryRelease & TestCategory!=ClientTelemetryEmulator & TestCategory!=Query & TestCategory!=ReadFeed & TestCategory!=Batch & TestCategory!=ChangeFeed & TestCategory!=LongRunning" --verbosity normal '
+
```
Comment (ealsur) [inline templates/build-test.yml:8]: Remove empty line

File: templates/build-test.yml [hunk 9]
```diff
@@ -5,10 +5,13 @@ parameters:
...
+  EmulatorPipeline1Arguments: ' --filter "TestCategory!=Quarantine & TestCategory!=Functional & TestCategory!=ClientTelemetryRelease & (TestCategory=ClientTelemetryEmulator|TestCategory=Query|TestCategory=ReadFeed|TestCategory=Batch|TestCategory=ChangeFeed)" --verbosity normal '
+  EmulatorPipeline2Arguments: ' --filter "TestCategory!=Quarantine & TestCategory!=Functional & TestCategory!=ClientTelemetryRelease & TestCategory!=ClientTelemetryEmulator & TestCategory!=Query & TestCategory!=ReadFeed & TestCategory!=Batch & TestCategory!=ChangeFeed" --verbosity normal '
+  EmulatorPipeline3Arguments: ' --filter "TestCategory!=Quarantine & TestCategory!=Functional & TestCategory!=ClientTelemetryRelease & (TestCategory=MultiRegion)" --verbosity normal '
```
Comment (ealsur) [inline templates/build-test.yml:11]:
```suggestion
  EmulatorPipeline3Arguments: ' --filter "TestCategory=MultiRegion" --verbosity normal '
```

File: templates/build-test.yml [hunk 10]
```diff
@@ -5,10 +5,13 @@ parameters:
   Arguments: ''
   VmImage: '' # https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops
   OS: 'Windows'
-  EmulatorPipeline1Arguments: ' --filter "TestCategory!=Quarantine & TestCategory!=Functional & TestCategory!=ClientTelemetryRelease & TestCategory!=LongRunning & (TestCategory=ClientTelemetryEmulator|TestCategory=Query|TestCategory=ReadFeed|TestCategory=Batch|TestCategory=ChangeFeed)" --verbosity normal '
-  EmulatorPipeline2Arguments: ' --filter "TestCategory!=Quarantine & TestCategory!=Functional & TestCategory!=ClientTelemetryRelease & TestCategory!=ClientTelemetryEmulator & TestCategory!=Query & TestCategory!=ReadFeed & TestCategory!=Batch & TestCategory!=ChangeFeed & TestCategory!=LongRunning" --verbosity normal '
+
+  EmulatorPipeline1Arguments: ' --filter "TestCategory!=Quarantine & TestCategory!=Functional & TestCategory!=ClientTelemetryRelease & (TestCategory=ClientTelemetryEmulator|TestCategory=Query|TestCategory=ReadFeed|TestCategory=Batch|TestCategory=ChangeFeed)" --verbosity normal '
```
Comment (ealsur) [inline templates/build-test.yml:9]: Undo all changes to the other EmulatorPipeline1/2Arguments

**Review by ealsur — APPROVED** [review]

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**